kuttl-test-upgrade.yaml

```diff
@@ -11,16 +11,13 @@
 # create a new one when the selector changed.
 apiVersion: kuttl.dev/v1beta1
 kind: TestSuite
-crdDir: ./tests/_build/crds/
 artifactsDir: ./tests/_build/artifacts/
 kindContainers:
-  - local/opentelemetry-operator:e2e
   - ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator:v0.49.0
 commands:
-  - command: make cert-manager
   - command: kubectl apply -f ./tests/e2e-upgrade/upgrade-test/opentelemetry-operator-v0.49.0.yaml
-  - command: kubectl rollout status -w deployment/opentelemetry-operator-controller-manager -n opentelemetry-operator-system
-  - command: sleep 60s
+  - command: kubectl wait --timeout=5m --for=condition=available deployment opentelemetry-operator-controller-manager -n opentelemetry-operator-system
+  - command: sleep 10
```

Comment:
why there is this sleep statement?

Reply:
It is due to the situation described in https://github.com/open-telemetry/opentelemetry-operator/pull/1254#discussion_r1037328129

When the deployment is done, if we create a OTEL Collector instance too fast, this error is thrown:
```
Internal error occurred: failed calling webhook "mopentelemetrycollector.kb.io": failed to call webhook: Post "[https://opentelemetry-operator-webhook-service.opentelemetry-operator-system.svc:443/mutate-opentelemetry-io-v1alpha1-opentelemetrycollector?timeout=10s](https://opentelemetry-operator-webhook-service.opentelemetry-operator-system.svc/mutate-opentelemetry-io-v1alpha1-opentelemetrycollector?timeout=10s)": dial tcp 10.96.63.229:443: connect: connection refused
```
Waiting a little bit, skips the error.

I was not able to find a better way to fix this... Because Kubernetes says the OTEL Operator deployment is ready. Do you know a better way to solve this?

Reply:
@iblancasa when the operator is ready and if you have some hints in the operator log, just wait for that. so you can remove this sleep.